### PR TITLE
feat: App details env filter

### PR DESCRIPTION
**Pull request by vivek-devtron**
# Description

This PR contains the changes of app environment filter

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Manually tested by creating multiple envs in same app

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/ApplicationGroup/AppGroupAppFilter.components.tsx [hunk 1]
```diff
@@ -25,7 +25,7 @@ export const ValueContainer = (props): JSX.Element => {
     } else {
         selectorText = `${selectedAppList.length > 0 ? selectedAppList.length : appListOptions.length}/${
             appListOptions.length
-        } Applications`
+        } ${filterParentType===FilterParentType.env? 'Applications': 'Environments'}`
```
Comment (shivani170) [inline src/components/ApplicationGroup/AppGroupAppFilter.components.tsx:28]: Please format this line

Comment (shivani170) [reply src/components/ApplicationGroup/AppGroupAppFilter.components.tsx:28]: if filterParentType===FilterParentType.env then are we showing the Application?

File: src/components/ApplicationGroup/AppGroupAppFilter.components.tsx [hunk 2]
```diff
@@ -152,6 +153,7 @@ export const MenuList = (props: any): JSX.Element => {
     const onTabChange = (e): void => {
         setSelectedFilterTab(e.currentTarget.dataset.selectedTab)
     }
+    const selectedType = filterParentType===FilterParentType.env? 'applications': 'environments'
```
Comment (shivani170) [inline src/components/ApplicationGroup/AppGroupAppFilter.components.tsx:156]: Format here as well

File: src/components/app/details/AppHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+            },
+        },
+        [appId, appName],
+    )
+
+    const renderAppDetailsTabs = () => {
+        return (
+            <ul role="tablist" className="tab-list">
+                <li className="tab-list__tab dc__ellipsis-right">
+                    <NavLink
```
Comment (shivani170) [inline src/components/app/details/AppHeader.tsx:121]: Can we keep all the tabs names in the array and iterate over that instead

File: src/components/app/details/cdDetails/CDDetails.tsx [hunk 4]
```diff
@@ -153,10 +153,15 @@ export default function CDDetails() {
...
             }
             if (result[0]) {
-                let _selectedEnvironment = (result[0]['value']?.result || []).find((envData) => {
-                    return +envId === envData.environmentId
-                })
-                setSelectedEnv(_selectedEnvironment)
+              const filteredEnvMap = filteredEnvIds?.split(',').reduce((agg, curr) => agg.set(curr, true), new Map())
+              let _selectedEnvironment = (result[0]['value']?.result || [])
+                  .filter((env) => {
+                    console.log(filteredEnvMap?.get(env.environmentId))
```
Comment (shivani170) [inline src/components/app/details/cdDetails/CDDetails.tsx:159]: Please remove the console

File: src/components/app/details/main.tsx [hunk 5]
```diff
@@ -27,19 +26,91 @@ const IndexComponent = lazy(() => import('../../v2/index'))
...
+            setSelectedGroupFilter([])
+            setAppListOptions([])
+        }
     }, [appId])
 
+    const getSavedFilterData = async (groupId?: number): Promise<void> => {
+        setSelectedAppList([])
+        setAppListLoading(true)
+        setGroupFilterOptions([])
+        // const { result } = await getEnvGroupList(+envId)
```
Comment (shivani170) [inline src/components/app/details/main.tsx:59]: Remove the commented code

File: src/components/app/details/main.tsx [hunk 6]
```diff
@@ -27,19 +26,91 @@ const IndexComponent = lazy(() => import('../../v2/index'))
...
+        // }
+        setAppListLoading(false)
+    }
+
+    const getAppListData = async (): Promise<void> => {
+        setSelectedAppList([])
+        setAppListLoading(true)
+        const { result } = await getAppOtherEnvironmentMin(appId)
+        console.log(result)
+        //setAppGroupListData(result)
```
Comment (shivani170) [inline src/components/app/details/main.tsx:98]: Remove this commented code

File: src/components/app/details/main.tsx [hunk 7]
```diff
@@ -54,9 +125,83 @@ export default function AppDetailsPage({ isV2 }: AppDetailsProps) {
...
+            }
+        } else {
+            for (const selectedApp of selectedAppList) {
+                selectedAppsMap[selectedApp.value] = true
+            }
+        }
+        for (const app of appListOptions) {
+            _allAppList.push({ id: app.value, appName: app.label, isSelected: selectedAppsMap[app.value] })
+        }
+        //setClickedGroup(_selectedGroup)
```
Comment (shivani170) [inline src/components/app/details/main.tsx:149]: Remove this commented code

File: src/components/app/details/cdDetails/CDDetails.tsx [hunk 8]
```diff
@@ -177,21 +182,26 @@ export default function CDDetails() {
...
-                    label: envData.environmentName,
-                    pipelineId: cdPipelinesMap.get(envData.environmentId)?.id,
-                    deploymentAppDeleteRequest: envData.deploymentAppDeleteRequest,
-                }
-            })
-            
+            const filteredEnvMap = filteredEnvIds?.split(',').reduce((agg, curr) => agg.set(+curr, true), new Map())
+            const envOptions: CICDSidebarFilterOptionType[] = (result[0]['value']?.result || [])
+            .filter((env) => {
+              console.log(filteredEnvMap?.get(env.environmentId))
```
Comment (shivani170) [inline src/components/app/details/cdDetails/CDDetails.tsx:188]: Please remove the console

File: src/components/ApplicationGroup/AppGroupAppFilter.components.tsx (hunk 1 above)
Comment (vivek-devtron) [reply src/components/ApplicationGroup/AppGroupAppFilter.components.tsx:28]: Done

File: src/components/ApplicationGroup/AppGroupAppFilter.components.tsx (hunk 2 above)
Comment (vivek-devtron) [reply src/components/ApplicationGroup/AppGroupAppFilter.components.tsx:156]: Done

File: src/components/app/details/AppHeader.tsx (hunk 3 above)
Comment (vivek-devtron) [reply src/components/app/details/AppHeader.tsx:121]: That is existing code so not going to do any major change there as this is urgent requirement. can be picked in next refactor.

File: src/components/app/details/cdDetails/CDDetails.tsx (hunk 4 above)
Comment (vivek-devtron) [reply src/components/app/details/cdDetails/CDDetails.tsx:159]: done

File: src/components/app/details/main.tsx (hunk 5 above)
Comment (vivek-devtron) [reply src/components/app/details/main.tsx:59]: this will be used once API is ready

File: src/components/app/details/main.tsx (hunk 7 above)
Comment (vivek-devtron) [reply src/components/app/details/main.tsx:149]: These will be used

File: src/components/app/details/cdDetails/CDDetails.tsx (hunk 8 above)
Comment (vivek-devtron) [reply src/components/app/details/cdDetails/CDDetails.tsx:188]: Done

**Review by shivani170 — APPROVED** [review]

**Pull request merged**